types/bluebird/index.d.ts

```diff
@@ -776,9 +776,16 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
   /**
    * Returns a function that can use `yield` to run asynchronous code synchronously. This feature requires the support of generators which are drafted in the next version of the language. Node version greater than `0.11.2` is required and needs to be executed with the `--harmony-generators` (or `--harmony`) command-line switch.
    */
-  // TODO fix coroutine GeneratorFunction
-  static coroutine(generatorFunction: (...args: any[]) => IterableIterator<any>): (...args: any[]) => Bluebird<any>;
-
+  // TODO: improve this to infer the return type of the generator, I believe this is blocked by https://github.com/Microsoft/TypeScript/issues/2983
```

Comment:
Please state _why_ it's blocked by https://github.com/Microsoft/TypeScript/issues/2983.

Something like

```
// TODO: After https://github.com/Microsoft/TypeScript/issues/2983 is fixed, we can use
// the return type propagation of generators to automatically infer the return type T.
```

ought to be enough.